src/event-engine/states/handshake_reconnecting.ts

```diff
@@ -19,34 +32,63 @@ export const HandshakeReconnectingState = new State<HandshakeReconnectingStateCo
 );
 
 HandshakeReconnectingState.onEnter((context) => handshakeReconnect(context));
-HandshakeReconnectingState.onExit(() => reconnect.cancel);
+HandshakeReconnectingState.onExit(() => handshakeReconnect.cancel);
 
-HandshakeReconnectingState.on(reconnectingSuccess.type, (context, event) =>
-  ReceivingState.with(
+HandshakeReconnectingState.on(handshakeReconnectSuccess.type, (context, event) => {
+  const cursor = {
+    timetoken: context?.timetoken ?? event.payload.cursor.timetoken,
+    region: event.payload.cursor.region,
+  };
+  return ReceivingState.with(
     {
       channels: context.channels,
       groups: context.groups,
-      cursor: event.payload.cursor,
+      cursor: cursor,
     },
-    [emitEvents(event.payload.events)],
-  ),
-);
+    [emitStatus({ category: categoryConstants.PNConnectedCategory })],
+  );
+});
 
-HandshakeReconnectingState.on(reconnectingFailure.type, (context, event) =>
+HandshakeReconnectingState.on(handshakeReconnectFailure.type, (context, event) =>
   HandshakeReconnectingState.with({ ...context, attempts: context.attempts + 1, reason: event.payload }),
 );
 
-HandshakeReconnectingState.on(reconnectingGiveup.type, (context) =>
-  HandshakeFailureState.with({
-    groups: context.groups,
-    channels: context.channels,
-    reason: context.reason,
-  }),
+HandshakeReconnectingState.on(handshakeReconnectGiveup.type, (context, event) =>
+  HandshakeFailedState.with(
+    {
+      groups: context.groups,
+      channels: context.channels,
+      timetoken: context.timetoken,
+      reason: event.payload,
+    },
+    [emitStatus({ category: categoryConstants.PNConnectionErrorCategory, error: event.payload?.message })],
+  ),
 );
 
 HandshakeReconnectingState.on(disconnect.type, (context) =>
   HandshakeStoppedState.with({
     channels: context.channels,
     groups: context.groups,
+    cursor: {
+      timetoken: context.timetoken ?? '0',
+      region: 0,
```

Comment:
Do we really need to strip away `region`?

Reply:
now it access incoming/context value.